app/models/user.rb

```diff
@@ -10,10 +10,10 @@ class User < ActiveRecord::Base
   has_many :authentications, dependent: :destroy
   has_many :cards, through: :decks
 
-  with_options if: lambda { |obj| obj.password.nil? } do |u|
-    u.validates :password, length: { minimum: 3 }, on: [:create, :update]
-    u.validates :password, confirmation: true, on: [:create, :update]
-    u.validates :password_confirmation, presence: true, on: [:create, :update]
+  with_options on: [:create, :update], if: lambda { |obj| obj.password.nil? } do |u|
```

Comment:
Line is too long. [84/80]<br>Use the new lambda literal syntax `->(params) {...}`.